src/utils/util.ts

```diff
@@ -88,45 +87,25 @@ export const isHtmlFile = (filePath: string) => {
   return hasFileExtension(filePath, ['html', 'htm']);
 };
 
-export const generatePreamble = (
-  config: d.Config,
-  opts: { prefix?: string; suffix?: string; defaultBanner?: boolean } = {}
-) => {
-  let preamble: string[] = [];
-
-  if (config.preamble) {
-    preamble = config.preamble.split('\n');
-  }
-
-  if (typeof opts.prefix === 'string') {
-    opts.prefix.split('\n').forEach((c) => {
-      preamble.push(c);
-    });
-  }
-
-  if (opts.defaultBanner === true) {
-    preamble.push(BANNER);
-  }
+/**
+ * Generate the preamble to be placed atop the main file of the build
+ * @param config the Stencil configuration file
+ * @return the generated preamble
+ */
+export const generatePreamble = (config: d.Config): string => {
+  const { preamble } = config;
 
-  if (typeof opts.suffix === 'string') {
-    opts.suffix.split('\n').forEach((c) => {
-      preamble.push(c);
-    });
+  if (preamble === undefined || preamble.length === 0) {
```

Comment:
Setting `preamble: null` in the config causes this to throw `TypeError: Cannot read properties of null (reading 'length') at generatePreamble`. I believe that was previously guarded against by:
https://github.com/ionic-team/stencil/blob/6987e4321b9dfd10710aa27a55e53e983e867729/src/utils/util.ts#L97

Reply:
Good call, fixed in https://github.com/ionic-team/stencil/pull/3085/commits/fe71934a2a19170bf3fde5b133a1df0283526eb1